pkg/frontend/openshiftcluster_putorpatch.go

```diff
@@ -233,3 +238,29 @@ func (f *frontend) _putOrPatchOpenShiftCluster(ctx context.Context, log *logrus.
 	}
 	return b, err
 }
+
+// stompSystemData will selectively overwrite systemData fields based on
+// arm inputs
+func stompSystemData(doc *api.OpenShiftClusterDocument, systemData *api.SystemData) {
+	if systemData == nil {
+		return
+	}
+	if systemData.CreatedAt != nil {
```

Comment:
I'm wondering if we don't want to stomp `CreatedAt`, `CreatedBy`, and `CreatedByType` as they ___should___ only be sent during the first creation by ARM, but in case they aren't the systemData for creation may be modified.  

```suggestion
 	if systemData.CreatedAt != nil && doc.OpenShiftCluster.SystemData.CreatedAt == nil {
```

Honestly might be a bad suggestion as we should only expect good behavior from ARM, just thinking though.

Reply:
I think I would like to trust ARM here to do good thing and dont have "what if somebody does something bad" logic. I think We are covered from bad things with pointers already enough?